stdlib/public/core/Integers.swift

```diff
@@ -1300,6 +1315,94 @@ extension BinaryInteger {
   }
 
+  @inlinable
+  public func isPower(of base: Self) -> Bool {
+    // Fast path when base is 2 or 10.
+    if base == 2 { return self._isPowerOfTwo }
+    if base == 10 { return self._isPowerOfTen }
+    // Slow path for other bases.
+    return self._isPower(of: base)
+  }
+  
+  @usableFromInline @_transparent
+  internal var _isPowerOfTwo: Bool {
+    return self > 0 && self & (self - 1) == 0
+  }
+
+  @usableFromInline
+  internal var _isPowerOfTen: Bool {
+    let exponent = self.trailingZeroBitCount
+    switch exponent {
+    case 0:  return self == 1 as UInt8
+    case 1:  return self == 10 as UInt8
+    case 2:  return self == 100 as UInt8
+    case 3:  return self == 1000 as UInt16
+    case 4:  return self == 10000 as UInt16
+    case 5:  return self == 100000 as UInt32
+    case 6:  return self == 1000000 as UInt32
+    case 7:  return self == 10000000 as UInt32
+    case 8:  return self == 100000000 as UInt32
+    case 9:  return self == 1000000000 as UInt32
+    case 10: return self == 10000000000 as UInt64
+    case 11: return self == 100000000000 as UInt64
+    case 12: return self == 1000000000000 as UInt64
+    case 13: return self == 10000000000000 as UInt64
+    case 14: return self == 100000000000000 as UInt64
+    case 15: return self == 1000000000000000 as UInt64
+    case 16: return self == 10000000000000000 as UInt64
+    case 17: return self == 100000000000000000 as UInt64
+    case 18: return self == 1000000000000000000 as UInt64
+    case 19: return self == 10000000000000000000 as UInt64
+    default:
+      // If this is 64-bit or less we can't have a higher power of 10
+      if self.bitWidth <= 64 { return false }
+
+      // Quickly check if parts of the bit pattern fits the power of 10.
+      //
+      // 10^0                                     1
+      // 10^1                                  1_01_0
+      // 10^2                               1_10_01_00
+      // 10^3                             111_11_01_000
+      // 10^4                          100111_00_01_0000
+      // 10^5                        11000011_01_01_00000
+      // 10^6                      1111010000_10_01_000000
+      // 10^7                   1001100010010_11_01_0000000
+      // 10^8                 101111101011110_00_01_00000000
+      // 10^9               11101110011010110_01_01_000000000
+      // 10^10           10010101000000101111_10_01_0000000000
+      // ...
+      // Column 1 is some "gibberish", which cannot be checked easily
+      // Column 2 is always the last two bits of the exponent
+      // Column 3 is always 01
+      // Column 4 is the trailing zeros, in equal number to the exponent value
+      //
+      // We check if Column 2 matches the last two bits of the exponent and
+      // Column 3 matches 0b01.
+      guard (self >> exponent)._lowWord & 0b1111 ==
+        ((exponent << 2) | 0b01) & 0b1111 else { return false }
+
+      // Now time for the slow path.
+      return self._isPower(of: 10)
```

Comment:
`(self >> exponent)._isPower(of: 5)`?

Reply:
Thank you for the prompt comments, Jordan!

I think this is not a sufficient condition, because it only ensures self == 2^exponent * 5^m, where exponent may not equal m. To make it correct, we'd do something like:
```swift
var powerOfFive = 1
for _ in 0..<exponent { powerOfFive *= 5 } // TODO: handle overflow here
return (self >> exponent) == powerOfFive
```

Reply:
Ugh, right, sorry, I jumped ahead to that in my head and then decided it wasn't necessarily worth it.